### PR TITLE
Auto import kubernetes template in Helm charts

### LINKED ISSUE #3265: Automatically import Kubernetes template when installing with Helm
### Problem
In #2267 a helm chart was added for a simpler install experience of the Coder dashboard. Teams who use the helm chart to install Coder to k8s will also likely want a "v1" style experience where new workspaces would be provisioned as K8s containerized workspaces.

### Definition of Done
We should "auto-import" https://github.com/coder/coder/tree/main/examples/templates/kubernetes-multi-service so that when a new Coder cluster is installed via the Helm chart, users can immediately create k8s containerized workspaces without further set-up required by admin.

### PR DESCRIPTION
**Pull request by deansheather**
- Adds new template `kubernetes` which is just a basic, single-container pod
- Adds new flag `--auto-import-template` (`CODER_TEMPLATE_AUTOIMPORT`) which can specify `kubernetes` and/or `kubernetes-multi-service`
    - After the first user (and org) is created, this will automatically create templates based on default template variables or information detected from the environment
        - For example, if we detect we're running in K8s then we use the namespace from the serviceaccount and use SA auth instead of kubeconfig auth
- Adds tests for the auto import template stuff
- Adds an SA to the helm chart
- Adds new helm value `coder.serviceAccount.workspacePerms` which controls whether to give the new SA perms to do workspace-related things

Closes #3265

### CODE REVIEW

File: examples/templates/kubernetes/main.tf [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+data "coder_workspace" "me" {}
+
+resource "coder_agent" "main" {
+  os   = "linux"
+  arch = "amd64"
+}
+
+resource "kubernetes_persistent_volume_claim" "home" {
+  metadata {
+    name      = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}-home"
```
Comment (bpmct) [inline examples/templates/kubernetes/main.tf:59]: Kubernetes validation fails with this

```sh
Error: metadata.0.name a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

File: examples/templates/kubernetes/main.tf [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        storage = "${var.home_size}Gi"
+      }
+    }
+  }
+}
+
+resource "kubernetes_pod" "main" {
+  count = data.coder_workspace.me.start_count
+  metadata {
+    name      = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
```
Comment (bpmct) [inline examples/templates/kubernetes/main.tf:75]: Kubernetes validation fails with this

```sh
Error: metadata.0.name a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

File: examples/templates/kubernetes/main.tf (hunk 1 above)
Comment (bpmct) [reply examples/templates/kubernetes/main.tf:59]: Edit: I realized my username was `Ben` so this is a problem, but not related to this newly-created template.

File: examples/templates/kubernetes/main.tf (hunk 2 above)
Comment (bpmct) [reply examples/templates/kubernetes/main.tf:75]: Edit: I realized my username was `Ben` so this is a problem, but not related to this newly-created template.

**Review by bpmct — COMMENTED** [review]
> Adds new flag --auto-import-template (CODER_TEMPLATE_AUTOIMPORT) which can specify kubernetes and/or kubernetes-multi-service

Do you specify multiple templates separated by commas?

We'll also want to add this to the helm docs before merging 👍🏼 

The auto-import worked great! ~~I ran into issues with the auto import~~

**Comment by deansheather** [comment]
> @ammario added this to the EE milestone

Does this mean you want this to be an enterprise only feature? Right now it's licensed AGPL...

**Comment by bpmct** [comment]
@deansheather nope, this one will be free to start. EE represents both free and paid features.

**Review by bpmct — COMMENTED** [review]
ran into an issue with the helm chart

> Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(Deployment.spec): unknown field "serviceAccountName" in io.k8s.api.apps.v1.DeploymentSpec

auto-import worked well, haven't been able to create a workspace yet though

**Comment by deansheather** [comment]
@bpmct works now!
image
image

**Review by f0ssel — APPROVED** [review]

**Pull request merged**